lightning/src/lib.rs

```diff
@@ -143,8 +143,16 @@ mod prelude {
 	pub use alloc::string::ToString;
 }
 
+#[cfg(all(feature = "std", test))]
+mod debug_sync;
+#[cfg(all(feature = "backtrace", feature = "std", test))]
+extern crate backtrace;
+
 #[cfg(feature = "std")]
 mod sync {
+	#[cfg(test)]
```

Comment:
Could enable it for `_test_utils` feature so it runs in other crates' tests

Reply:
That's....more complicated than it looks. That changes our public API to use our order-checking locks instead of std::mutex, which means we have to expose them and update downstream code to use the right lock in the right place, and std in other places. I figured it wasn't worth the complexity after fighting with a few variations initially.